pandas/io/excel/_base.py

```diff
@@ -739,7 +722,13 @@ def __init__(
             ext = os.path.splitext(path)[-1]
             self.check_extension(ext)
 
-        self.path = path
+        if "b" not in mode:
+            mode += "b"
+        mode = mode.replace("a", "r+")
+
+        self.handles = IOHandles(path, compression={"copression": None})
+        if not isinstance(path, ExcelWriter):
+            self.handles = get_handle(path, mode, is_text=False)
```

Comment:
`to_excel` does not take the `storage_options` argument. It would make sense to add it (`read_excel` takes this argument).

Reply:
yep